py-polars/tests/unit/io/test_file_cache.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+import pytest
+
+import polars as pl
+
+
+@pytest.mark.slow()
+def test_file_cache_ttl(
+    monkeypatch: pytest.MonkeyPatch, io_files_path: Path, tmp_path: Path
+) -> None:
+    tmp_path = tmp_path.absolute()
+    io_files_path = io_files_path.absolute()
+    monkeypatch.setenv("POLARS_FORCE_ASYNC", "1")
+    monkeypatch.setenv("POLARS_TEMP_DIR", str(tmp_path))
+
+    from time import sleep
+
+    from blake3 import blake3
+
+    paths = [
+        io_files_path / "foods1.csv",
+        io_files_path / "foods2.csv",
+    ]
+    hashes = [blake3(bytes(x)).hexdigest()[:32] for x in paths]
+    metadata_file_paths = [tmp_path / f"file-cache/m/{x}" for x in hashes]
+    data_file_dir = tmp_path / "file-cache/d/"
+    pl.scan_csv(paths[0]).collect()
+    pl.scan_csv(paths[1], file_cache_ttl=0).collect()
+
+    has_data_file = [False, False]
+    for data_file in data_file_dir.iterdir():
+        for i, h in enumerate(hashes):
+            has_data_file[i] |= data_file.name.startswith(h)
+
+    assert all(x.exists() for x in metadata_file_paths)
```

Comment:
This test works locally, but fails in CI - there is some path difference that I'm not sure how to address

(https://github.com/pola-rs/polars/actions/runs/9478367898/job/26114665783?pr=16891)

Reply:
A few things come to mind:

- When using `tmp_path` it can help to call `tmp_path.mkdir(exist_ok=True)` before actually using it. Not sure why that is. You can try it.
- Why do you need `tmp_path = tmp_path.absolute()`? I haven't seen this in any other tests.
- Not sure why this only pops up in async reader tests, but that probably has something to do with it...

Reply:
Ah - thanks for the hint! I think I know what's going on - the static variable holding the temporary directory path must have been initialized to the default by the one of the previous tests because we run everything with FORCE_ASYNC